snippet-extractor-output/snippets.yaml

```diff
@@ -10456,20 +10456,44 @@
 
     const id = $("#notificationId").val().toString();
 
-    const details = 
-      {
+
+    let details;
+
+    // Identifies whether the current mail item is in read or compose mode.
+
+    if (Office.context.mailbox.item.itemId != undefined) {
+      // Read mode.
+      details = {
+        type: Office.MailboxEnums.ItemNotificationMessageType.InsightMessage,
+        message: "This is an insight notification with id = " + id,
+        icon: "icon1",
+        actions: [
+          {
+            actionText: "Open insight",
+            actionType: Office.MailboxEnums.ActionType.ShowTaskPane,
+            commandId: "PG.HelpCommand.Read",
+            contextData: { a: "aValue", b: "bValue" }
+          }
+        ]
+      };
+    } else {
+      // Compose mode.
+      details = {
         type: Office.MailboxEnums.ItemNotificationMessageType.InsightMessage,
-        message: "This is an insight notification",
+        message: "This is an insight notification with id = " + id,
         icon: "icon1",
         actions: [
           {
             actionText: "Open insight",
             actionType: Office.MailboxEnums.ActionType.ShowTaskPane,
-            commandId: "msgComposeOpenPaneButton",
+            commandId: "PG.HelpCommand.Compose",
```

Comment:
This seems to be the only difference between the two `details` objects. So, can we scope the `if...else` check to just these two, which will make it clearer?

Reply:
I forgot that we can use the ternary operator within JS objects. Updated the code to use this for clarity.